demystifying_large_language_models/demystifying_large_language_models.md

```diff
@@ -0,0 +1,691 @@
+
+## LLM Costs and Risks
+
+Costs
+---
+
+Over time, with advances in the speed and size of computer processors (especially Graphical Processing Units, or GPUs), faster and more efficient data storage, and competition among cloud providers, computing has become less expensive.  That means that tasks can be performed today that would have been unrealistic (in terms of time and expense) to perform a few years ago.  That's true across computing, and it's certainly true in artificial intelligence.  But "less expensive" doesn't mean free.  Large language models are incredibly resource intensive, in terms of the amount of input required, the amount of computing required, and the energy consumed in building and running the models. 
+
+
+Risks
+---
+
+Additionally, there are important risks to consider.  We're going to borrow a framework from Weidinger et al.  In their superb and approachable ["Ethical and social risks of harm from Language Models,"](https://arxiv.org/abs/2112.04359) Weidinger and her co-authors enumerate six domains of risks, harms, and costs associated with large language models.  We summarize these risks below (and note, we have avoided the temptation to ask a large language model to summarize the text for us):
+
+1. Discrimination, exclusion, and toxicity
+-----
+
+Where can one obtain huge amounts of text, generated by humans, for free?  The internet!  Language models and their abilities to predict text have grown in part because of better, faster, and cheaper computers, but also in part because of the massive proliferation of human-generated text that can be analyzed.  Unfortunately, this set of language is biased in terms of who generates it and the opinions they share.  Toxic language, stereotypes, and bias abound, and these have already been documented in text generated by LLMs.  
+
+2. Information Hazards
+------
+
+Because the training data given to LLMs is so vast, there are cases (some involving litigation) of private information such as home addresses being leaked by LLMs.  Additionally, LLMs might correctly infer and disclose sensitive information while amassing and summarizing data (for example, accurately disclosing a serious disease diagnosis to an unprepared user asking about their symptoms).
```

Comment:
```suggestion
Because the training data given to LLMs is so vast, there are cases of private information such as home addresses being leaked by LLMs.  Additionally, LLMs might correctly infer and disclose sensitive information while amassing and summarizing data (for example, accurately disclosing a serious disease diagnosis to an unprepared user asking about their symptoms).
```

Reply:
Ah, I totally get wanting to highlight legal risks -- I think for me, this quick paranthetical didn't read as enough focus to actually be highlighting legal risks, so it felt more to me just like additional, but overall unnecessary, information about the fact that this is something that has already been observed. 

If we want to highlight legal risks, which seems valuable, I'm thinking it might make sense to more explicitly highlight them? Also, the legal risk here in particular seems to be more a risk to the LLM developers, rather than to the LLM end users, to whom this module is aimed. 

I might be misunderstanding the kinds of cases that have been brought forward, though! If there are cases that are being brought against users for giving someone else's sensitive data to a conversational agent, and then that data winding up in the training data, and then subsequently the model, then I think we should _definitely_ very explicitly mention that as a legal risk, maybe even in its own section, or at least small para?

Reply:
I think you're on to something with the legal risk being to LLM developers here... I'll omit!